.github/workflows/test.yml

```diff
@@ -0,0 +1,60 @@
+name: 'Test'
+on:
+  pull_request:
+    branches: [ "master" ]
+  push:
+    branches: [ "master" ]
+  workflow_dispatch:
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+
+  integration-tests:
+    name: "Integration tests"
+    runs-on: [self-hosted, linux, normal]
+    steps:
+      - name: 'Check out code'
+        uses: actions/checkout@v4
+        with:
+          # Check out pull request HEAD instead of merge commit.
+          ref: ${{ github.event.pull_request.head.sha }}
+          submodules: recursive
+
+      - name: "Set up nightly Rust" # https://github.com/rust-lang/rustup/issues/3409
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: nightly-2024-08-28
+
+      - name: 'Set up tree for rust dependency'
+        run: make setup
+
+      - name: 'Cache smir_pretty and rustc'
+        uses: Swatinem/rust-cache@v2
+        with:
+          workspaces: |
+            .
+            deps/rust/src
+          cache-directories: |
+            target
+            deps/rust/src/build
+            deps/rust/src/target
+
+      - name: 'Build smir_pretty and its rustc dependency'
+        run: | # rustc bootstrap checks this and refuses stage 1 in "CI"
+          export GITHUB_ACTIONS="in denial" && \
+          echo "GITHUB_ACTIONS = ${GITHUB_ACTIONS}" && \
+          make build_all
+
+      - name: 'Run smir integration tests'
+        run: |
+          make integration-test
+
+      - name: 'Clean up toolchain'
+        if: always()
+        run: |
+          make rustup-clear-toolchain
```

Comment:
This is important, otherwise the build will leave a dangling toolchain behind that `rustup` stumbles upon during setup. I [opened a ticket for this against `dtolnay/rust-toolchain@`](https://github.com/dtolnay/rust-toolchain/issues/127).

Reply:
Looks like a solution is already merged @jberthold